pcsx2/gui/FrameForGS.cpp

```diff
@@ -183,13 +185,19 @@ void GSPanel::DoResize()
 	else if( arr > 1 )
 		viewport.y = (int)( (double)viewport.y/arr + 0.5);
 
+	if (g_Conf->GSWindow.ScaleToImageSize && image_width > 0 && image_width < client.GetWidth() &&
+		image_height > 0 && image_height < client.GetHeight() && g_Conf->GSWindow.AspectRatio != AspectRatio_Stretch) {
+			viewport.x = image_width;
+			viewport.y = (int)((double)image_width / targetAr);
+	}
+
 	float zoom = g_Conf->GSWindow.Zoom.ToFloat()/100.0;
-	if( zoom == 0 )//auto zoom in untill black-bars are gone (while keeping the aspect ratio).
+	if( zoom == 0 ) // auto zoom in until black bars are gone (while keeping the aspect ratio).
 		zoom = std::max( (float)arr, (float)(1.0/arr) );
 
 	viewport.Scale(zoom, zoom*g_Conf->GSWindow.StretchY.ToFloat()/100.0 );
 	if (viewport == client && EmuConfig.Gamefixes.FMVinSoftwareHack && g_Conf->GSWindow.IsFullscreen)
-		viewport.x += 1; //avoids crash on some systems switching HW><SW in fullscreen aspect ratio's with FMV Software switch.
+		viewport.x += 1; // avoids crash on some systems switching HW><SW in fullscreen aspect ratio's with FMV Software switch.
```

Comment:
Seems like you are a seer https://github.com/PCSX2/pcsx2/pull/1312#issuecomment-217485166
I wonder if it's still actual though.